recipes/librdkafka/all/conanfile.py

```diff
@@ -75,51 +73,49 @@ def build_requirements(self):
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+    def layout(self):
+        cmake_layout(self)
+        self.folders.source = "src"
+        self.folders.build = "build"
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["WITHOUT_OPTIMIZATION"] = self.settings.build_type == "Debug"
-        cmake.definitions["ENABLE_DEVEL"] = self.settings.build_type == "Debug"
-        cmake.definitions["RDKAFKA_BUILD_STATIC"] = not self.options.shared
-        cmake.definitions["RDKAFKA_BUILD_EXAMPLES"] = False
-        cmake.definitions["RDKAFKA_BUILD_TESTS"] = False
-        cmake.definitions["WITHOUT_WIN32_CONFIG"] = True
-        cmake.definitions["WITH_BUNDLED_SSL"] = False
-        cmake.definitions["WITH_ZLIB"] = self.options.zlib
-        cmake.definitions["WITH_ZSTD"] = self.options.zstd
-        cmake.definitions["WITH_PLUGINS"] = self.options.plugins
-        cmake.definitions["WITH_SSL"] = self.options.ssl
-        cmake.definitions["WITH_SASL"] = self.options.sasl
-        cmake.definitions["ENABLE_LZ4_EXT"] = True
-        if tools.Version(self.version) >= "1.9.0":
-            cmake.definitions["WITH_CURL"] = self.options.curl
-
-        cmake.configure()
-        return cmake
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["WITHOUT_OPTIMIZATION"] = self.settings.build_type == "Debug"
+        tc.variables["ENABLE_DEVEL"] = self.settings.build_type == "Debug"
+        tc.variables["RDKAFKA_BUILD_STATIC"] = not self.options.shared
+        tc.variables["RDKAFKA_BUILD_EXAMPLES"] = False
+        tc.variables["RDKAFKA_BUILD_TESTS"] = False
+        tc.variables["WITHOUT_WIN32_CONFIG"] = True
+        tc.variables["WITH_BUNDLED_SSL"] = False
+        tc.variables["WITH_ZLIB"] = self.options.zlib
+        tc.variables["WITH_ZSTD"] = self.options.zstd
+        tc.variables["WITH_PLUGINS"] = self.options.plugins
+        tc.variables["WITH_SSL"] = self.options.ssl
+        tc.variables["WITH_SASL"] = self.options.sasl
+        tc.variables["ENABLE_LZ4_EXT"] = True
+        if Version(self.version) >= "1.9.0":
+            tc.variables["WITH_CURL"] = self.options.curl
+        tc.generate()
 
     def build(self):
-        self._patch_sources()
-        with tools.run_environment(self):
-            cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        with tools_legacy.environment_append({"PKG_CONFIG_PATH": self.generators_folder}):
+            cmake.configure()
             cmake.build()
```

Comment:
I've tested all shared (even pkgconf in build requirements) on macOS with `librdkafka:sasl=True` to test pkgconf, and it works like a charm thanks to `VirtualBuildEnv`, no SIP issue!